ksql-functional-tests/src/main/java/io/confluent/ksql/test/tools/TestCase.java

```diff
@@ -294,4 +329,244 @@ private static String standardizeTopology(final String topology) {
     return result.toString();
   }
 
+  @SuppressWarnings("unchecked")
+  static void processSingleRecord(
+      final FakeKafkaRecord fakeKafkaRecord,
+      final FakeKafkaService fakeKafkaService,
+      final TopologyTestDriverContainer testDriver,
+      final KafkaTopicClient kafkaTopicClient,
+      final SchemaRegistryClient schemaRegistryClient) throws IOException, RestClientException {
+    final Serializer serializer = fakeKafkaRecord.getTestRecord().topic.getSerdeSupplier()
+        instanceof AvroSerdeSupplier
```

Comment:
There are a bunch of very similar-looking ternary statements for serializer/record switches such as this one throughout the code. It'd be nice to have a helper function perform this switch.

Reply:
Good point. I'll refactor them in the follow up PRs.